components/inputs/test/input-text.vdiff.js

```diff
@@ -0,0 +1,173 @@
+		{ name: 'unit-rtl', template: unitFixture, rtl: true },
+		{ name: 'unit-disabled', template: html`<d2l-input-text label="unit" label-hidden value="value" unit="%" disabled></d2l-input-text>` },
+		{ name: 'unit-invalid', template: unitInvalidFixture },
+		{ name: 'unit-invalid-focus', template: unitInvalidFixture, focus: true },
+		{ name: 'unit-invalid-rtl', template: unitInvalidFixture, rtl: true },
+		{ name: 'unit-invalid-rtl-focus', template: unitInvalidFixture, focus: true, rtl: true },
+		{
+			name: 'unit-init-hidden',
+			template: html`<d2l-input-text label="unit" label-hidden value="overflowing value should not overlap unit" unit="%" style="display: none;"></d2l-input-text>`,
+			action: async(elem) => elem.style.display = 'inline-block'
+		},
+		{
+			name: 'unit-change',
+			template: html`<d2l-input-text label="unit" label-hidden value="3" value-align="end" unit="/5" unit-label="out of 5"></d2l-input-text>`,
+			action: async(elem) => { elem.unit = '/5000'; await elem.updateComplete; }
+		},
+		{ name: 'override-height', template: html`<d2l-input-text label="Name" label-hidden value="text" style="--d2l-input-height: 100px;"></d2l-input-text>` },
+		{ name: 'override-padding', template: html`<d2l-input-text label="Name" label-hidden value="text" style="--d2l-input-padding: 2rem 3rem;--d2l-input-padding-focus: calc(2rem - 1px) calc(3rem - 1px);"></d2l-input-text>` },
+		{ name: 'override-text-align', template: html`<d2l-input-text label="Name" label-hidden value="text" style="--d2l-input-text-align: end;"></d2l-input-text>` },
+		{ name: 'icon-left', template: iconLeftFixture },
+		{ name: 'icon-left-rtl', template: iconLeftFixture, rtl: true },
+		{ name: 'icon-right', template: iconRightFixture },
+		{ name: 'icon-right-rtl', template: iconRightFixture, rtl: true },
+		{
+			name: 'icon-left-label',
+			template: html`
+				<d2l-input-text label="Name" value="Lorem ipsum dolor sit amet, consectetur adipiscing elit">
+					${createIcon('calendar', 'left')}
+				</d2l-input-text>
+			`
+		},
+		{
+			name: 'icon-right-label',
+			template: html`
+				<d2l-input-text label="Name" value="Lorem ipsum dolor sit amet, consectetur adipiscing elit">
+					${createIcon('calendar', 'right')}
+				</d2l-input-text>
+			`
+		},
+		{
+			name: 'icon-left-right',
+			template: html`
+				<d2l-input-text label="Name" label-hidden value="Lorem ipsum dolor sit amet, consectetur adipiscing elit">
+					${createIcon('search', 'left')}
+					${createIcon('search', 'right')}
+				</d2l-input-text>
+			`
+		},
+		{
+			name: 'button-icon-left',
+			template: html`
+				<d2l-input-text label="Name" label-hidden value="Lorem ipsum dolor sit amet, consectetur adipiscing elit">
+					${createIcon('search', 'left')}
+				</d2l-input-text>
+			`
+		},
+		{
+			name: 'button-icon-right',
+			template: html`
+				<d2l-input-text label="Name" label-hidden value="Lorem ipsum dolor sit amet, consectetur adipiscing elit">
```

Comment:
I added hidden labels to a lot of these as errors were getting thrown, causing subsequent tests to fail. Haha, so not _quite_ isolated.